### ISSUE TITLE
Assets name collision in dev mode

### ISSUE THREAD

**Issue by adrienbaron** [nuxt/nuxt.js#3786]
### Version

[next](https://github.com/nuxt.js/releases/tag/next)

### Reproduction link

[https://github.com/adrienbaron/nuxt-asset-name-collision-bug](https://github.com/adrienbaron/nuxt-asset-name-collision-bug)

### Steps to reproduce

- Run the provided Nuxt app in dev mode

### What is expected ?

- Images should be different (one big, one small)

### What is actually happening?

- Images are the same

Nuxt seems to copy all the images to `/_nuxt/img`, but doesn't add a hash of the file at the end of the name. So images with the same name but in different folders collides.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c7617">#c7617</a>)</em></sub></div>

**Comment by clarkdo** [comment]
Currently Nuxt will flat all built resources by default.

With #3787, you can config `build.filenames.img` as what you want, for example:


```js
      build: {
        filenames: {
          img: (file) => {
            const dir = path.dirname(file).replace('assets', 'img')
            return `${dir}/[name].[hash:7].[ext]`
          }
        }
      }
```

**Comment by adrienbaron** [comment]
@clarkdo Doesn't seem to resolve the issue, the issue only happens in dev mode, in regular build mode the hashing is applied and there is no problem.

**Comment by clarkdo** [comment]
It because in `dev` mode, we removed all `hash` because it has some problems in `https://github.com/webpack/webpack/issues/1914#issuecomment-174171709`

If you use the function filename, if will generate images into different directories.

**Comment by adrienbaron** [comment]
@clarkdo Ok makes sense. Using the path in dev mode seems like a good compromise? Would still store if the user moves files around probably though... Not sure how much this is an issue.
Worse case we could just disallow files with same name in assets and add a warning in the doc?

**Comment by clarkdo** [comment]
Fixed by #3789
